ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py

```diff
@@ -24,54 +23,52 @@
     "integer",
     "signed_integer",
     "complex",
-    "real_and_complex",
-    "float_and_integer",
-    "float_and_complex",
     "bool",
 }
 
 
+def get_dtype_combinations(keys):
+    return set(["_and_".join(combination)
+                for r in range(1, len(keys) + 1)
+                for combination in combinations(keys, r)])
+
+
+_dtype_kind_keys = _dtype_kind_keys.union(get_dtype_combinations(_dtype_kind_keys))
+
+
 def _get_fn_dtypes(framework, kind="valid"):
     return test_globals.CURRENT_RUNNING_TEST.supported_device_dtypes[framework.backend][
         test_globals.CURRENT_DEVICE_STRIPPED
     ][kind]
 
 
-def _get_type_dict(framework, kind):
-    if kind == "valid":
-        return framework.valid_dtypes
-    elif kind == "numeric":
-        return framework.valid_numeric_dtypes
-    elif kind == "integer":
-        return framework.valid_int_dtypes
-    elif kind == "float":
-        return framework.valid_float_dtypes
-    elif kind == "unsigned":
-        return framework.valid_uint_dtypes
-    elif kind == "signed_integer":
-        return tuple(
-            set(framework.valid_int_dtypes).difference(framework.valid_uint_dtypes)
-        )
-    elif kind == "complex":
-        return framework.valid_complex_dtypes
-    elif kind == "real_and_complex":
-        return tuple(
-            set(framework.valid_numeric_dtypes).union(framework.valid_complex_dtypes)
-        )
-    elif kind == "float_and_complex":
-        return tuple(
-            set(framework.valid_float_dtypes).union(framework.valid_complex_dtypes)
-        )
-    elif kind == "float_and_integer":
-        return tuple(
-            set(framework.valid_float_dtypes).union(framework.valid_int_dtypes)
-        )
-    elif kind == "bool":
-        return tuple(
-            set(framework.valid_dtypes).difference(framework.valid_numeric_dtypes)
-        )
-    else:
-        raise RuntimeError("{} is an unknown kind!".format(kind))
+def _get_type_dict(framework, *kinds):
```

Comment:
Wouldn't it be more convenient to pass the `kind` as a list instead of processing a string? I think it will be more efficient and easier to use.